### PR TITLE
Evolution

### PR DESCRIPTION
**Pull request by charm-13**
Implemented evolution capabilities for creatures.

### CODE REVIEW

File: src/api/creatures.py [hunk 1]
```diff
@@ -238,8 +286,56 @@ def adopt_creature(user_id: int, creature_id: int):
...
+                LEFT JOIN user_creature_connection ON creature_id = creatures.id 
+                    AND user_id = :u_id
+                JOIN evolution_stages ON creatures.stage = evolution_stages.stage
+                WHERE creatures.id = :c_id
+            """), 
+            {"u_id": user_id, "c_id": creature_id}).mappings().fetchone()
+            
+            print(f"{status}")
+            
+            if status == None:
```
Comment (thuang-28) [inline src/api/creatures.py]: can be written as "if not status"

File: src/api/creatures.py [hunk 2]
```diff
@@ -238,8 +286,56 @@ def adopt_creature(user_id: int, creature_id: int):
...
+                return {"success": False, "error": f"Creature {creature_id} does not exist."}
+            
+            if not status["is_adopted"]:
+                return {"success": False, "error": f"Creature {creature_id} must already be adopted before evolving."}
+            
+            if status["stage"] == 3:
+                return {"success": False, "error": f"Creature {creature_id} is already at the highest stage."}
+            
+            if status["hunger"] < status["max_hunger"] or status["happiness"] < status["max_happiness"]:
+                return {"success": False, "error": f"The creature must have maximum stats before evolving."}
```
Comment (thuang-28) [inline src/api/creatures.py]: For consistency, add creature id to this error message as well

File: APISpec.md [hunk 3]
```diff
@@ -157,6 +159,18 @@ Adopts a creature. User's affinity level with the specified creature must be 100
...
 
+### 2.5 Evolve Creatures `/users/{user_id}/creatures/{creature_id}/evolve` (POST)
+
+Each creature can be 1 of 3 stages. This evolves a creature to the next stage. 
+
+**Response:**
+
+```json
+{
+    "success": "boolean"
```
Comment (samikshakarimbil) [inline APISpec.md]: It might help to return what stage the creature was evolved to

**Review by samikshakarimbil — CHANGES_REQUESTED** [review]
Overall looks good, I agree with Tracy's feedback

**Review by samikshakarimbil — APPROVED** [review]
Looks good to me

**Review by thuang-28 — APPROVED** [review]
yea, looks good to me

**Pull request merged**